------------------------------------------------------------------------
r64 | wensong | 2010-07-22 01:50:58 +0200 (Thu, 22 Jul 2010) | 11 lines

This fixes what appears to be a typo in the netlink client code
whereby the activeconns is read as activeconns and then
overwritten by persistentconns. The result is that the active
connection count reported by ipvsadm -L is wrong if the
code is compiled to use the netlink interface.

Tested-by: Nigel Kukard <nkukard@lbsd.net>
Signed-off-by: Simon Horman <horms@verge.net.au>
Signed-off-by: Wensong Zhang <wensong@linux-vs.org>


------------------------------------------------------------------------

Index: ipvsadm/libipvs/libipvs.c
===================================================================
--- ipvsadm/libipvs/libipvs.c	(revision 63)
+++ ipvsadm/libipvs/libipvs.c	(revision 64)
@@ -748,7 +748,7 @@
 	d->entrytable[i].l_threshold = nla_get_u32(dest_attrs[IPVS_DEST_ATTR_L_THRESH]);
 	d->entrytable[i].activeconns = nla_get_u32(dest_attrs[IPVS_DEST_ATTR_ACTIVE_CONNS]);
 	d->entrytable[i].inactconns = nla_get_u32(dest_attrs[IPVS_DEST_ATTR_INACT_CONNS]);
-	d->entrytable[i].activeconns = nla_get_u32(dest_attrs[IPVS_DEST_ATTR_PERSIST_CONNS]);
+	d->entrytable[i].persistconns = nla_get_u32(dest_attrs[IPVS_DEST_ATTR_PERSIST_CONNS]);
 	d->entrytable[i].af = d->af;
 
 	if (ipvs_parse_stats(&(d->entrytable[i].stats),
